Application/EdFi.Ods.Extensions.TPDM/Artifacts/MsSql/Data/Security/1010-TPDM-ResourceClaims.sql

```diff
@@ -1,465 +1,6178 @@
-);
+BEGIN
+    DECLARE 
+        @applicationId AS INT,
+        @claimId AS INT,
+        @claimName AS nvarchar(max),
+        @parentResourceClaimId AS INT,
+        @existingParentResourceClaimId AS INT,
+        @claimSetId AS INT,
+        @claimSetName AS nvarchar(max),
+        @authorizationStrategyId AS INT,
+        @msg AS nvarchar(max),
+        -- Well-known values
+        @createActionId AS INT = 1,
+        @readActionId AS INT = 2,
+        @updateActionId AS INT = 3,
+        @deleteActionId AS INT = 4
+
+    DECLARE @claimIdStack AS TABLE (Id INT IDENTITY, ResourceClaimId INT)
+
+    SELECT @applicationId = ApplicationId
+    FROM [dbo].[Applications] WHERE ApplicationName = 'Ed-Fi ODS API';
+
+    BEGIN TRANSACTION
+
+    -- Push claimId to the stack
+    INSERT INTO @claimIdStack (ResourceClaimId) VALUES (@claimId)
+
+    -- Processing children of root
+    ----------------------------------------------------------------------------------------------------------------------------
+    -- Resource Claim: 'http://ed-fi.org/ods/identity/claims/domains/tpdm'
+    ----------------------------------------------------------------------------------------------------------------------------
+    SET @claimName = 'http://ed-fi.org/ods/identity/claims/domains/tpdm'
+    SET @claimId = NULL
+
+    SELECT @claimId = ResourceClaimId, @existingParentResourceClaimId = ParentResourceClaimId
+    FROM dbo.ResourceClaims 
+    WHERE ClaimName = @claimName
+
+    SELECT @parentResourceClaimId = ResourceClaimId
+    FROM @claimIdStack
+    WHERE Id = (SELECT Max(Id) FROM @claimIdStack)
+
+    IF @claimId IS NULL
+        BEGIN
+            PRINT 'Creating new claim: ' + @claimName
+
+            INSERT INTO dbo.ResourceClaims(DisplayName, ResourceName, ClaimName, ParentResourceClaimId, Application_ApplicationId)
+            VALUES ('tpdm', 'tpdm', 'http://ed-fi.org/ods/identity/claims/domains/tpdm', @parentResourceClaimId, @applicationId)
+
+            SET @claimId = SCOPE_IDENTITY()
+        END
+    ELSE
+        BEGIN
+            IF @parentResourceClaimId != @existingParentResourceClaimId
+            BEGIN
+                PRINT 'Repointing claim ''' + @claimName + ''' (ResourceClaimId=' + CONVERT(nvarchar, @claimId) + ') to new parent (ResourceClaimId=' + CONVERT(nvarchar, @parentResourceClaimId) + ')'
+
+                UPDATE dbo.ResourceClaims
+                SET ParentResourceClaimId = @parentResourceClaimId
+                WHERE ResourceClaimId = @claimId
+            END
+        END
+  
+    -- Processing claim sets for http://ed-fi.org/ods/identity/claims/domains/tpdm
+    ----------------------------------------------------------------------------------------------------------------------------
+    -- Claim set: 'Ed-Fi Sandbox'
+    ----------------------------------------------------------------------------------------------------------------------------
+    SET @claimSetName = 'Ed-Fi Sandbox'
+    SET @claimSetId = NULL
```

Comment:
this is redundant. this is a valid assumption that the sandbox claim set will be available.

Reply:
It just adds unnecessary complexity to the XSLT style sheet to eliminate it. It's ultimately a question of what we want to optimize for. I'm putting forth a direction of moving away from manual maintenance of SQL scripts, so this redundancy wouldn't be much of a concern (other than possibly due to the "bloat" of the script) and keeping the style sheet simpler is better, so long as it is not causing any problems.